### ISSUE TITLE
Pre-training

### ISSUE THREAD

**Issue by cassinius** [cassinius/knowledge-cards#4]
I am grateful for *finally understanding `pre-training`*!!! It's an `unsupervised` technique to make a neural network learn the `general feature distribution` of a data set, thus *biasing* it towards the real-world distribution before the training phase. This generally reduces variance and thereby `reduces overfitting near local optima`. The network is trained layer-by-layer, meaning that all previously trained layers are fixed in their weight vectors. Eventually, the network can also be *fine-tuned* by running a normal supervised feed-forward / back-propagation round (I don't understand WHY...!?). It was also mentioned that pre-training could help in `weakly/semi-supervised` learning, but I didn't get how this would work (apart from the fact that the pre-trained network would somehow `compute labels`...).
